pennylane/hf/hartree_fock.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for performing the self-consistent-field iterations.
+"""
+
+import autograd.numpy as anp
+from pennylane.hf.matrices import (
+    generate_core_matrix,
+    generate_overlap_matrix,
+    generate_repulsion_tensor,
+    molecular_density_matrix,
+)
+
+
+def generate_scf(mol, n_steps=50, tol=1e-8):
+    r"""Return a function that performs the self-consistent-field iterations.
```

Comment:
This is an important function, but the docstring is comparatively less informative than with other parts of the code. I suggest adding equations showing the self-consistent field equations and the definition of the quantities involved. I also recommend adding a short description of the iterative procedure that aims to calculate the self-consistent equations